modules/tracking/src/roiSelector.cpp

```diff
@@ -163,7 +163,7 @@ namespace cv {
     // while key is not ESC (27)
     while(key!=27){
       temp=select(windowName, img, true, fromCenter);
-      if(temp.width>0 && temp.height>0)
+      if(key!=27 && temp.width>0 && temp.height>0)
```

Comment:
Isn't this enforced by the while loop? It is a matter of micro seconds of difference at most ...